python-package/xgboost/core.py

```diff
@@ -369,22 +357,6 @@ def _maybe_dt_array(array):
     return array
 
 
-def _check_data(data, missing):
-    '''The missing value applies only to np.ndarray.'''
-    is_invalid = (not isinstance(data, np.ndarray)) and (missing is not None)
-    is_invalid = is_invalid and not math.isnan(missing)
-    if is_invalid:
-        raise ValueError(
-            'missing value only applies to dense input, ' +
-            'e.g. `numpy.ndarray`.' +
-            ' For a possibly sparse data type: ' + str(type(data)) +
-            ' please remove missing values or set it to nan.' +
-            ' Current missing value is set to: ' + str(missing))
-    if isinstance(data, list):
```

Comment:
So this warning about initialising from list is removed?

Reply:
@RAMitchell Let me just throw an error.